server/socket/socketHelperFunctions.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-unused-vars */
+require('express');
+const mongoose = require('mongoose');
+const ssotToRobotParser = require('../services/SsotToRobotParsing/SsotToRobotParser.js');
+const ssotModels = require('../models/singleSourceOfTruthModel.js');
+const userAccessObject = require('../models/userAccessObjectModel.js');
+const jobsModel = require('../models/robotJobModel.js');
+
+mongoose.set('useFindAndModify', false);
+
+/**
+ * @description Fetches the ssot from the database and parses the ssot to robot code
+ * @param {String} robotId the id of the robot we want the robot code for
+ */
+exports.getRobotCode = async (robotId) => {
+  try {
+    mongoose.connect(process.env.MONGODB_URI, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+    });
+    const ssot = await mongoose.model('SSoT').findById(robotId).exec();
+    const robotCode = ssotToRobotParser.parseSsotToRobotCode(ssot);
+    return robotCode;
+  } catch (err) {
+    return console.error(err);
+  }
+};
+
+/**
+ * @description Fetches the ssot from the database and parses the ssot to robot code
+ * @param {String} robotId the id of the robot we want the robot code for
+ */
+exports.getAllUserIds = async () => {
+  try {
+    mongoose.connect(process.env.MONGODB_URI, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true,
+    });
+    const userObjs = await mongoose.model('userAccessObject').find();
```

Comment:
Are you sure this works? Afaik this returns a query

Reply:
Yes, this works. .find() returns all the documents from the collection in an array